tests/test_routes.py

```diff
@@ -6,3 +9,38 @@ def test_index_page(flask_test_client):
 def test_upload_page(flask_test_client):
     response = flask_test_client.get("/upload")
     assert response.status_code == 200
+
+
+@patch("app.main.routes.post_ingest")
+def test_upload_xlsx_success(
+    mock_response, example_pre_ingest_data_file, flask_test_client
+):
+    mock_response.content = b'"Success: Spreadsheet data ingested"\n'
+    response = flask_test_client.post(
+        "/upload", data={"file-upload-1": example_pre_ingest_data_file}
+    )
+    assert response.status_code == 200
+
+
+@patch("app.main.routes.post_ingest")
+def test_upload_xlsx_errors(
+    mock_response, example_pre_ingest_data_file, flask_test_client
+):
+    mock_response.content = (
+        b'{"validation_errors": [\n "Wrong Type Failure: Sheet \\'
+        b'"Funding\\" column \\"Spend for Reporting Period\\" expected type'
+        b' \\"float64\\", got type \\"object\\"",\n "Enum Value Failure: '
+        b'Sheet \\"RiskRegister\\" Column \\"Pre-mitigatedImpact\\" Row 5 '
+        b'Value \\"Error\\" is not a valid enum value.",\n ]\n}\n'
+    )
+    response = flask_test_client.post(
+        "/upload", data={"file-upload-1": example_pre_ingest_data_file}
+    )
+    assert response.status_code == 200
+
+
+def test_upload_wrong_format(flask_test_client, example_ingest_wrong_format):
+    response = flask_test_client.post(
+        "/upload", data={"file-upload-1": example_ingest_wrong_format}
+    )
+    assert response.status_code == 500
```

Comment:
Can we return a `400` in this case rather than `500` as we should handle this case.